### PR TITLE
Feature/task instructions

### PR DESCRIPTION
**Pull request by nikita-dnestrov**

### CODE REVIEW

**Comment by kyrylo-burmelov** [comment]
[Task Instructions Layout](https://trello.com/c/3JvWk8ed/20-task-instructions-layout)

File: client/src/components/basic/task-instructions/components/Rank/rank.module.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	border: #f09712 solid 2px;
+	color: #f09712;
+}
+
+:local(.7),
+:local(.8),
+:local(.9) {
+	border: #ec4179 solid 2px;
+	color: #ec4179;
+}
```
Comment (kyrylo-burmelov) [inline client/src/components/basic/task-instructions/components/Rank/rank.module.scss:28]: class name can not start from number

File: client/src/components/basic/task-instructions/components/Rank/index.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import styles from './rank.module.scss';
+
+interface IRankProps {
+	difficulty: number;
+}
+
+const Rank: React.FC<IRankProps> = (props) => {
```
Comment (kyrylo-burmelov) [inline client/src/components/basic/task-instructions/components/Rank/index.tsx:8]: Let's us existing Rank component. Check from develop branch /basic

File: client/src/components/basic/task-instructions/index.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	favorites: number;
+	rating: number;
+	createdBy: string;
+}
+
+const TaskInstructions: React.FC<ITaskInstructionProps> = (props: ITaskInstructionProps) => {
+	return (
+		<div className={styles.taskContainer}>
+			<div className={styles.header}>
+				<Rank difficulty={6} />
```
Comment (kyrylo-burmelov) [inline client/src/components/basic/task-instructions/index.tsx:19]: You should not mock data like that. You need to mock all the data from container, component should just display the data

File: client/src/components/basic/task-instructions/index.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+				<h2 className={styles.title}>{props.title}</h2>
+			</div>
+			<div className={styles.task}>
+				<div className={styles.switch}>
+					<a className={styles.active}>Instructions</a>
+					<a>Output</a>
```
Comment (kyrylo-burmelov) [inline client/src/components/basic/task-instructions/index.tsx:25]: It should work like tabs. You should be able to navigate between tabs by clicking. Output should be empty tab for now.

File: client/src/components/basic/task-instructions/index.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+				<Statistics
+					icon="timeline-bar-chart"
+					content={<p>{props.rating}% positive feedback</p>}
+					divider={true}
+				/>
+				<Statistics
+					icon="person"
+					content={
+						<p>
+							created by <a>Emerson Saris</a>
```
Comment (kyrylo-burmelov) [inline client/src/components/basic/task-instructions/index.tsx:41]: Same thing about mock data

File: client/src/containers/routing/index.tsx [hunk 6]
```diff
@@ -1,13 +1,45 @@
...
 	<Switch>
-		<PublicRoute exact restricted={false} path="/" component={Example} />
+		{/* <PublicRoute exact restricted={false} path="/" component={} /> */}
+		<TaskInstructions
+			title="Stop gninnipS My sdroW!"
+			html={html}
+			favorites={500}
+			rating={88}
+			createdBy="Emerson Saris"
+		/>
```
Comment (kyrylo-burmelov) [inline client/src/containers/routing/index.tsx:42]: You should not mock data like this. Let's create a container and mock data from container. Container should return component or several components.

```javascript
const mockData = {
   rating: 88,
   favorites: 500,
   /* .... */
}
```

See example how it should be from /users/:name route container.

You need to create route for your page too. Let it be /task/:id

File: client/src/components/basic/task-instructions/components/Rank/index.tsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (kyrylo-burmelov) [inline client/src/components/basic/task-instructions/components/Rank/index.tsx:1]: file or folder name should follow [JS quality criteria](https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/source/javascript.md#naming)

File: client/src/components/basic/task-instructions/components/Rank/rank.module.scss [hunk 8]
```diff
@@ -0,0 +1,28 @@
+:local(.rank) {
```
Comment (kyrylo-burmelov) [inline client/src/components/basic/task-instructions/components/Rank/rank.module.scss:1]: file or folder name should follow [JS quality criteria](https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/source/javascript.md#naming)

File: client/src/components/basic/task-instructions/components/Statistics/index.tsx [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
```
Comment (kyrylo-burmelov) [inline client/src/components/basic/task-instructions/components/Statistics/index.tsx:1]: file or folder name should follow [JS quality criteria](https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/source/javascript.md#naming)

File: client/src/components/basic/task-instructions/components/Statistics/statistics.module.scss [hunk 10]
```diff
@@ -0,0 +1,24 @@
+:local(.info) {
```
Comment (kyrylo-burmelov) [inline client/src/components/basic/task-instructions/components/Statistics/statistics.module.scss:1]: file or folder name should follow [JS quality criteria](https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/source/javascript.md#naming)

**Review by kyrylo-burmelov — CHANGES_REQUESTED** [review]

File: client/src/components/basic/task-instructions/components/Rank/index.tsx (hunk 2 above)
Comment (kyrylo-burmelov) [reply client/src/components/basic/task-instructions/components/Rank/index.tsx:8]: components/basic/rank

**Comment by nikita-dnestrov** [comment]
For some reason 'git merge develop' doesnt pull all files

**Comment by nikita-dnestrov** [comment]
I've got a few questions left:

1. How to handle and debug private route? For now I placed task instructions on public route.
2. Is it ok to place task instructions in basic folder or should it be more logical to place it in common or pages or whatever you see more appropriate.

**Review by kyrylo-burmelov — APPROVED** [review]

**Review by taras-dubyk — APPROVED** [review]

**Pull request merged**